### ISSUE TITLE
Allow summary stats for a given project and field

### ISSUE THREAD

**Issue by rmcolq** [CLIMB-TRE/onyx#63]
Enable some sort of summary about the contents of the database in a particular field e.g. for `adm1_country` would return a dictionary of country:count pairs. Ideally could be extendable to any filter command so for a given filter, could summarize keys and counts across one or more fields

**Comment by warrickball** [comment]
Can this issue be closed? It looks to me like it's resolved by #66. I just tried with version 2.4.0 of the client and got what I think is the behaviour you describe here. e.g.
```
$ onyx filter mscapetest --summarise sample_type --format csv
sample_type,count
other,7
stool,188
```

**Comment by warrickball** [comment]
I haven't checked the Python API behaviour, though.

**Comment by tombch** [comment]
With the Python API it is achieved similarly, taking an example from the [documentation for filter](https://climb-tre.github.io/onyx-client/api/documentation/client/#onyx.OnyxClient.filter):

```
import os
from onyx import OnyxConfig, OnyxEnv, OnyxClient

config = OnyxConfig(
    domain=os.environ[OnyxEnv.DOMAIN],
    token=os.environ[OnyxEnv.TOKEN],
)

with OnyxClient(config) as client:
    records = list(
        client.filter(
            project="project",
            fields={
                "field1": "abcd",
                "published_date__range" : "2023-01-01, 2023-01-02",
            },
            summarise="published_date",
        )
    )
```
Results in:
```
>>> records
[
    {
        "published_date": "2023-01-01",
        "count": 1,
    },
    {
        "published_date": "2023-01-02",
        "count": 1,
    },
]
```

**Comment by tombch** [comment]
Actually it will be reopened, due to:
`Ideally could be extendable to any filter command so for a given filter, could summarize keys and counts across one or more fields`

As this functionality doesn't exist yet.

**Comment by tombch** [comment]
Just an update: happy to say this functionality has been implemented on the development branch in https://github.com/CLIMB-TRE/onyx/commit/2257a1057de51d0dc2ad4d66db3c39335d132d5a ! So should be available soon :)

**Comment by tombch** [comment]
This has now been merged into the master branch and will be available in version 0.8.0
